src/iframe-helper.js

```diff
@@ -214,7 +215,8 @@ function registerGlobalListenerIfNeeded(parentWin) {
       return;
     }
 
-    let listeners = listenForEvents[data['type']];
```

Comment:
Unfortunately I don't think this will work either as only the opt_typekey of the first call to registerGlobalListenerIfNeeded will be referenced.   You could store the values you're listening for on a window variable and reference that way.  I suggest you reach out to AMP core and get their take as they will undoubtedly have an opinion on how best to proceed.

Reply:
Went with the approach of saving the event type keys to the window object. Not sure who to ask, but there's a good chance they're already reviewers on this PR.